### PR TITLE
[V8] Update paranets functionality

### PR DESCRIPTION
**Pull request by zsculac**

### CODE REVIEW

File: dkg/paranet.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def __init__(self, manager: DefaultRequestManager):
     _register_paranet = Method(BlockchainRequest.register_paranet)
 
     def create(
-        self, ual: UAL, name: str, description: str
+        self, ual: UAL, name: str, description: str, nodes_access_policy: int, miners_access_policy: int
```
Comment (u-hubar) [inline dkg/paranet.py:60]: Let's make `nodes_access_policy` and `miners_access_policy` Enums, take a look how it's done for other stuff in the `dataclasses.py`

File: dkg/paranet.py [hunk 2]
```diff
@@ -83,6 +85,318 @@ def create(
             "operation": json.loads(Web3.to_json(receipt)),
         }
 
+    _add_paranet_curated_nodes = Method(BlockchainRequest.add_paranet_curated_nodes)
+
+    def add_curated_nodes(
+        self, ual: UAL, identity_ids: list[int]
```
Comment (u-hubar) [inline dkg/paranet.py:91]:
```suggestion
        self, paranet_ual: UAL, identity_ids: list[int]
```

File: dkg/paranet.py [hunk 3]
```diff
@@ -83,6 +85,318 @@ def create(
...
+    ) -> dict[str, str | HexStr | TxReceipt]:
+        parsed_ual = parse_ual(ual)
+        knowledge_asset_storage, knowledge_asset_token_id = (
```
Comment (u-hubar) [inline dkg/paranet.py:94]:
```suggestion
        paranet_knowledge_asset_storage, paranet_knowledge_asset_token_id = (
```

File: dkg/paranet.py [hunk 4]
```diff
@@ -83,6 +85,318 @@ def create(
...
+            parsed_ual["contract_address"],
+            parsed_ual["token_id"],
+        )
+
+        self._add_paranet_curated_nodes(
+            knowledge_asset_storage,
```
Comment (u-hubar) [inline dkg/paranet.py:100]:
```suggestion
            paranet_knowledge_asset_storage,
```

File: dkg/paranet.py [hunk 5]
```diff
@@ -83,6 +85,318 @@ def create(
...
+            knowledge_asset_token_id,
```
Comment (u-hubar) [inline dkg/paranet.py:101]:
```suggestion
            paranet_knowledge_asset_token_id,
```

File: dkg/paranet.py [hunk 6]
```diff
@@ -83,6 +85,318 @@ def create(
...
+            identity_ids,
+        )
+
+        return {
+            "paranetUAL": ual,
+            "paranetId": Web3.to_hex(
+                Web3.solidity_keccak(
+                    ["address", "uint256"],
+                    [knowledge_asset_storage, knowledge_asset_token_id],
```
Comment (u-hubar) [inline dkg/paranet.py:110]:
```suggestion
                    [paranet_knowledge_asset_storage, paranet_knowledge_asset_token_id],
```

File: dkg/paranet.py [hunk 7]
```diff
@@ -83,6 +85,318 @@ def create(
...
+                )
+            ),
+            "operation": "Add curated nodes"
```
Comment (u-hubar) [inline dkg/paranet.py:113]: Operations for these should be tx receipts, look into other functions

File: dkg/paranet.py [hunk 8]
```diff
@@ -83,6 +85,318 @@ def create(
...
+            parsed_ual["token_id"],
+        )
+
+        paranet_id = Web3.solidity_keccak(
+            ["address", "uint256"], [knowledge_asset_storage, knowledge_asset_token_id]
+        )
+
+        curated_nodes = self._get_curated_nodes(paranet_id)
+
+        # curated_nodes is a list of ABIDecodedNamedTuple type so it needs to be converted to a dictionary
```
Comment (u-hubar) [inline dkg/paranet.py:236]: Can we perhaps make it smarter somehow?

**Review by u-hubar — CHANGES_REQUESTED** [review]

File: dkg/paranet.py (hunk 7 above)
Comment (zsculac) [reply dkg/paranet.py:118]: Yes but in this case we do not have tx receipts. Perhaps I should just remove it from the return?

File: dkg/paranet.py (hunk 8 above)
Comment (zsculac) [reply dkg/paranet.py:241]: Not sure, I tried a couple solutions already. This is the simplest one I could come up with if we want to have a response look like in dkg.js:
```
{
    "0": 1,
    "1": "0x516d577966336474714a6e687543707a4544544e6d4e46596335746a7854725868476355556d474864673267746a",
    "identityId": 1,
    "nodeId": "0x516d577966336474714a6e687543707a4544544e6d4e46596335746a7854725868476355556d474864673267746a"
}
```

Comment (u-hubar) [reply dkg/paranet.py:259]: We don't need it to be the same, this is not pythonic way

File: dkg/paranet.py (hunk 7 above)
Comment (u-hubar) [reply dkg/paranet.py]: We have receipts for all transactions, should have just add variable to store the returned receipts

**Pull request merged**